### ISSUE TITLE
please clarify difference between autoencoder loss and reconstruction loss - warning messages concerning losses and parameters in Parametric UMAP

### ISSUE THREAD

**Issue by gg4u** [lmcinnes/umap#1035]
Please refer to:
https://github.com/lmcinnes/umap/blob/master/umap/parametric_umap.py#L98

I am confused about the type of losses:

```
        encoder : tf.keras.Sequential, optional
            The encoder Keras network
        decoder : tf.keras.Sequential, optional
            the decoder Keras network
        parametric_embedding : bool, optional
            Whether the embedder is parametric or non-parametric, by default True
        parametric_reconstruction : bool, optional
            Whether the decoder is parametric or non-parametric, by default False
        parametric_reconstruction_loss_fcn : bool, optional
            What loss function to use for parametric reconstruction, by default tf.keras.losses.BinaryCrossentropy
        parametric_reconstruction_loss_weight : float, optional
            How to weight the parametric reconstruction loss relative to umap loss, by default 1.0
        autoencoder_loss : bool, optional
            [description], by default False
        reconstruction_validation : array, optional
            validation X data for reconstruction loss, by default None
        loss_report_frequency : int, optional
            how many times per epoch to report loss, by default 1
        n_training_epochs : int, optional
            number of epochs to train for, by default 1
```
I am using autoencoding UMAP models:
So far I understood that `parametric_reconstruction_loss_fcn` is the reconstruction loss of an autoencoder ( when passing encoder and decoder models).

It is default BinaryCrossentropy, but I am passing MSE (I am using data inputs in range [-1, 1]):

```
parametric_reconstruction_loss_fcn = tf.keras.losses.MSE, 
```

In spite of passing MSE, I always receive a warningn:

```
UserWarning: Data should be scaled to the range 0-1 for cross-entropy reconstruction loss.
```

1. why that warning? is the MSE actually taken ?
2. Is the parametric reconstruction loss the reconstruction loss between decoder and encoder, right ?
So what is the autoencoder_loss ?
https://github.com/lmcinnes/umap/blob/master/doc/parametric_umap.rst
https://github.com/lmcinnes/umap/blob/a7606f2d413c278cc98d932de62f829914209c4f/notebooks/Parametric_UMAP/04.0-parametric-umap-mnist-embedding-convnet-with-autoencoder-loss.ipynb#L211
